policy-controller/k8s/index/src/outbound/index/http.rs

```diff
@@ -100,36 +130,24 @@ fn convert_linkerd_rule(
         .map(convert_linkerd_filter)
         .collect::<Result<_>>()?;
 
-    let request_timeout = rule.timeouts.as_ref().and_then(|timeouts| {
-        let timeout = time::Duration::from(timeouts.request?);
+    timeouts.request = timeouts.request.or_else(|| {
+        rule.timeouts.as_ref().and_then(|timeouts| {
+            let timeout = time::Duration::from(timeouts.request?);
 
-        // zero means "no timeout", per GEP-1742
-        if timeout == time::Duration::from_nanos(0) {
-            return None;
-        }
+            // zero means "no timeout", per GEP-1742
+            if timeout == time::Duration::from_nanos(0) {
```

Comment:
nit: there's a const we can use

```suggestion
            if timeout == time::Duration::ZERO {
```